ext/bg/js/util.js

```diff
@@ -26,6 +26,47 @@ function utilIsolate(data) {
     return JSON.parse(JSON.stringify(data));
 }
 
+function utilSetEqual(setA, setB) {
+    if (setA.size !== setB.size) {
+        return false;
+    }
+
+    for (const value of setA) {
+        if (!setB.has(value)) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+function utilSetIntersection(setA, setB) {
+    return new Set(
+        [...setA].filter(value => setB.has(value))
+    );
+}
+
+function utilSetDifference(setA, setB) {
+    return new Set(
+        [...setA].filter(value => !setB.has(value))
+    );
+}
+
+function utilStringHashCode(string) {
+    let hashCode = 0;
+
+    if (string.length === 0) {
```

Comment:
This conditional is unnecessary since if `string.length == 0` the for loop will not execute and you will end up return `0` anyway.